src/sinks/aws_cloudwatch_logs/sink.rs

```diff
@@ -0,0 +1,96 @@
+use crate::event::{Event, EventFinalizers, Finalizable};
+
+use crate::sinks::aws_cloudwatch_logs::request_builder::{
+    CloudwatchRequest, CloudwatchRequestBuilder,
+};
+use crate::sinks::aws_cloudwatch_logs::retry::CloudwatchRetryLogic;
+use crate::sinks::aws_cloudwatch_logs::service::{CloudwatchLogsPartitionSvc, CloudwatchResponse};
+use crate::sinks::aws_cloudwatch_logs::CloudwatchKey;
+
+use crate::sinks::util::service::Svc;
+use crate::sinks::util::SinkBuilderExt;
+
+use async_trait::async_trait;
+
+use futures::future;
+use futures::stream::BoxStream;
+use futures::StreamExt;
+
+use vector_core::buffers::{Ackable, Acker};
+use vector_core::partition::Partitioner;
+use vector_core::sink::StreamSink;
+use vector_core::stream::BatcherSettings;
+
+pub struct CloudwatchSink {
+    pub batcher_settings: BatcherSettings,
+    pub request_builder: CloudwatchRequestBuilder,
+    pub acker: Acker,
+    pub service: Svc<CloudwatchLogsPartitionSvc, CloudwatchRetryLogic<CloudwatchResponse>>,
+}
+
+impl CloudwatchSink {
+    async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let request_builder = self.request_builder;
+        let batcher_settings = self.batcher_settings;
+        let service = self.service;
+        let acker = self.acker;
+
+        input
+            .filter_map(|event| {
+                future::ready(match request_builder.build(event) {
+                    Ok(maybe_req) => maybe_req.map(Ok),
+                    Err(err) => Some(Err(err)),
+                })
+            })
```

Comment:
This could be:

```rust
.filter_map(|event| async { request_builder.build(event).transpose() })
```

Reply:
fixed in ad628ee0140d02cb601f09e1145031ea7bd15895